### PR TITLE
Array queue in js and ts

### PR DESCRIPTION
**Pull request by S-N-O-R-L-A-X**
Array queue in js and ts

### CODE REVIEW

File: codes/javascript/chapter_stack_and_queue/array_queue.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
+/**
+ * File: array_queue.js
+ * Created Time: 2022-12-13
+ * Author: S-N-O-R-L-A-X (snorlax.xu@outlook.com)
+ */
+
+
+/* 基于环形数组实现的队列 */
+class ArrayQueue {
+    queue;     // 用于存储队列元素的数组
```
Comment (krahets) [inline codes/javascript/chapter_stack_and_queue/array_queue.js:10]: Use private member, `#queue`

File: codes/javascript/chapter_stack_and_queue/array_queue.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    front = 0;  // 头指针，指向队首
+    rear = 0;   // 尾指针，指向队尾 + 1
```
Comment (krahets) [inline codes/javascript/chapter_stack_and_queue/array_queue.js:12]: Same as above

File: docs/chapter_stack_and_queue/queue.md [hunk 3]
```diff
@@ -718,13 +718,203 @@ comments: true
...
+    console.log("队列是否为空 = " + empty);
+
+    /* 测试环形数组 */
+    for (let i = 0; i < 10; i++) {
+        queue.offer(i);
+        queue.poll();
+        console.log("第 " + i + " 轮入队 + 出队后 queue = ");
+        console.log(queue.toArray());
+    }
+
```
Comment (krahets) [inline docs/chapter_stack_and_queue/queue.md:917]: Redundant empty line.

**Comment by krahets** [comment]
Thanks for the contribution! This PR will keep open waiting for the `linkedlist_queue` implementation.

By the way, could you help implement `linkedlist_stack.js`? As `array_stack.js` and `stack.js` are written by you as well. Maybe it is proper for you to finish them. Thanks!

**Comment by S-N-O-R-L-A-X** [comment]
> 

As I said before, I was not sure I would have enough time in the following days. So if you are willing to wait, I will spend time complementing it in the future.

**Comment by krahets** [comment]
Yes, it is recommended to finish at least one markdown file in one pull request.

The `list.js/ts` and `array_list.js/ts` belong to `stack.md` and have been merged in the past pull request, it would be better to work on `linkedlist_stack.js` to complete the whole `stack.md`.

This regulation is helpful for other contributors to recognize "which file they can work on" as I've marked you as the author of JS, TS code of the `stack.md`.

Thanks!

**Comment by S-N-O-R-L-A-X** [comment]
> Yes, it is recommended to finish at least one markdown file in one pull request.
> 
> The `list.js/ts` and `array_list.js/ts` belong to `stack.md` and have been merged in the past pull request, it would be better to work on `linkedlist_stack.js` to complete the whole `stack.md`.
> 
> This regulation is helpful for other contributors to recognize "which file they can work on" as I've marked you as the author of JS, TS code of the `stack.md` in #15 .
> 
> Thanks!

That's OK. I was not so clear about the file structture before.

**Comment by S-N-O-R-L-A-X** [comment]
@krahets Hi, I have completed queue but I suddenly find that this title is Array queue. So should we still use this to merge the linkedlist_queue? Or should we change the title?

**Comment by krahets** [comment]
Prefer to changing the title.

**Comment by S-N-O-R-L-A-X** [comment]
> Prefer to changing the title.

It is easy to change the title for this PR. But the branch is also called array_queue, is it easy to rename a branch?

**Comment by krahets** [comment]
I think it is not necessary. We can ignore this problem this time.

**Comment by S-N-O-R-L-A-X** [comment]
It seems going wrong when merging another branch to this branch.

**Comment by krahets** [comment]
> It seems going wrong when cherry-picking my commits from another branch to this branch.

Please close this PR and re-create one.

**Comment by krahets** [comment]
This PR is closed because of the inconsistent header with the master branch of the main repo.

It is recommended to 

1. save your work locally
2. re-fork the repo.
3. Copy your code to the repo.
4. Create a pull request.

**Comment by S-N-O-R-L-A-X** [comment]
> This PR is closed because of the inconsistent header with the master branch of the main repo.
> 
> It is recommended to
> 
> 1. save your work locally
> 2. re-fork the repo.
> 3. Copy your code to the repo.
> 4. Create a pull request.

I used another branch and reraised a PR.

**Pull request closed without merging**